### PR TITLE
[MTDSA-6848] - Reliefs RAML alignments

### PR DESCRIPTION
**Pull request by Thepworth95**

### CODE REVIEW

File: resources/public/api/conf/1.0/schemas/amend_relief_other_request.json [hunk 1]
```diff
@@ -54,7 +54,7 @@
           "example": "INPOLY123A"
         },
         "amount": {
-          "description": "The amount of relief claimed on redemption of bonus shares.",
+          "description": "The amount of relief claimed on qualifying shares and securities.",
```
Comment (Mbrookesy) [inline resources/public/api/conf/1.0/schemas/amend_relief_other_request.json:57]: This change needs to also be made for retrieve other reliefs

**Review by Mbrookesy — CHANGES_REQUESTED** [review]

**Review by Mbrookesy — APPROVED** [review]

File: resources/public/api/conf/1.0/application.raml [hunk 2]
```diff
@@ -83,7 +83,7 @@ uses:
         uriParameters:
           taxYear:
             description: 'The tax year the data applies to, for example, 2021-22. The tax year must have ended. The start year and end year must not span two tax years. The minimum tax year is 2021-22. No gaps are allowed, for example, 2020-22 is not valid. (The minimum tax year in Sandbox is 2019-20.)'
-            example: "2019-20"
+            example: "2021-22"
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/application.raml:86]: Pensions isn't part of this change - and in the code, there's nothing specifying the minimum year in sandbox vs other environments. Something to discuss after stand up tomorrow?

File: resources/public/api/conf/1.0/errors.raml [hunk 3]
```diff
@@ -91,6 +91,61 @@ traits:
...
                   code: FORMAT_VALUE
+  formatNameExSpouse:
+    responses:
+      400:
+        body:
+          application/json:
+            type: types.errorResponse
+            examples:
+              formatNameExSpouse:
+                description: The format of exSpouseName is not valid
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/errors.raml:102]: Full stops after each error description.

File: resources/public/api/conf/1.0/errors.raml [hunk 4]
```diff
@@ -91,6 +91,61 @@ traits:
...
+                  code: FORMAT_NAME_EX_SPOUSE
+  formatNameBusiness:
+    responses:
+      400:
+        body:
+          application/json:
+            type: types.errorResponse
+            examples:
+              formatNameBusiness:
+                description: The format of businessName is not valid
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/errors.raml:113]: Full stops after each error description.

File: resources/public/api/conf/1.0/errors.raml [hunk 5]
```diff
@@ -91,6 +91,61 @@ traits:
...
+                  code: FORMAT_NAME_BUSINESS
+  formatNatureOfTrade:
+    responses:
+      400:
+        body:
+          application/json:
+            type: types.errorResponse
+            examples:
+              formatNatureOfTrade:
+                description: The format of natureOfTrade is not valid
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/errors.raml:124]: Full stops after each error description.

File: resources/public/api/conf/1.0/errors.raml [hunk 6]
```diff
@@ -91,6 +91,61 @@ traits:
...
+                  code: FORMAT_NATURE_OF_TRADE
+  formatIncomeSource:
+    responses:
+      400:
+        body:
+          application/json:
+            type: types.errorResponse
+            examples:
+              formatIncomeSource:
+                description: The format of incomeSource is not valid
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/errors.raml:135]: Full stops after each error description.

File: resources/public/api/conf/1.0/errors.raml [hunk 7]
```diff
@@ -91,6 +91,61 @@ traits:
...
+                  code: FORMAT_INCOME_SOURCE
+  formatLenderName:
+    responses:
+      400:
+        body:
+          application/json:
+            type: types.errorResponse
+            examples:
+              formatLenderName:
+                description: The format of lenderName is not valid
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/errors.raml:146]: Full stops after each error description.

File: resources/public/api/conf/1.0/reliefInvestments_amend.raml [hunk 8]
```diff
@@ -4,6 +4,7 @@ is:
   - headers.testScenarioHeader
   - errors.formatNino
   - errors.formatTaxYear
+  - errors.ruleTaxYearNotSupported
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/reliefInvestments_amend.raml:7]: missing FORMAT_UNIQUE_INVESTMENT_REFERENCE, FORMAT_NAME

File: resources/public/api/conf/1.0/schemas/amend_relief_investments_request.json [hunk 9]
```diff
@@ -37,7 +37,8 @@
           }
         },
         "required": [
-          "uniqueInvestmentRef"
+          "uniqueInvestmentRef",
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/schemas/amend_relief_investments_request.json:40]: uniqueInvestmentRef missing regex

File: resources/public/api/conf/1.0/schemas/amend_relief_investments_request.json [hunk 10]
```diff
@@ -80,7 +81,9 @@
...
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/schemas/amend_relief_investments_request.json:84]: uniqueInvestmentRef missing regex

File: resources/public/api/conf/1.0/schemas/amend_relief_investments_request.json [hunk 11]
```diff
@@ -118,7 +121,8 @@
...
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/schemas/amend_relief_investments_request.json:124]: uniqueInvestmentRef missing regex

File: resources/public/api/conf/1.0/schemas/amend_relief_investments_request.json [hunk 12]
```diff
@@ -156,7 +160,8 @@
...
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/schemas/amend_relief_investments_request.json:163]: uniqueInvestmentRef missing regex

File: resources/public/api/conf/1.0/schemas/amend_relief_investments_request.json [hunk 13]
```diff
@@ -194,7 +199,8 @@
...
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/schemas/amend_relief_investments_request.json:202]: uniqueInvestmentRef missing regex

File: resources/public/api/conf/1.0/schemas/amend_relief_other_request.json [hunk 14]
```diff
@@ -54,7 +54,7 @@
           "example": "INPOLY123A"
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/schemas/amend_relief_other_request.json:54]: All customerReference missing regex

Comment (james-work-account) [reply resources/public/api/conf/1.0/schemas/amend_relief_other_request.json:54]: Same for natureOfTrade

File: resources/public/api/conf/1.0/schemas/retrieve_relief_investments_response.json [hunk 15]
```diff
@@ -37,7 +42,8 @@
           }
         },
         "required": [
-          "uniqueInvestmentRef"
+          "uniqueInvestmentRef",
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/schemas/retrieve_relief_investments_response.json:45]: uniqueInvestmentRef missing regex in all places

File: resources/public/api/conf/1.0/schemas/retrieve_relief_others_response.json [hunk 16]
```diff
@@ -131,7 +136,7 @@
           }
         },
         "required": [
-          "customerReference"
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/schemas/retrieve_relief_others_response.json:134]: customerReference and missing regex in all places

**Review by james-work-account — CHANGES_REQUESTED** [review]
Add regex to a field with something like this:
```
"pattern": "^[0-9a-zA-Z{À-˿’}\\- _&`():.'^]{1,90}$"
```
Although these regexes don't show up on the documentation preview, they stop invalid JSON examples (there are a few invalid fields in here which you should across by adding this).

File: resources/public/api/conf/1.0/application.raml (hunk 2 above)
Comment (Thepworth95) [reply resources/public/api/conf/1.0/application.raml:86]: from memory its already in pentions, and its one of caseys endpoints not nicks and nick said he was going to talk to casey about when he was going to do any alignments required on that when hes back from his A/L

**Review by james-work-account — APPROVED** [review]

**Pull request merged**